### PR TITLE
Settings levels and timer fallback epg

### PR DESCRIPTION
**Pull request by phunkyfish**
v3.23.0
- Added: Support settings levels via the current kodi settings level
- Added: Fallback EPG Entries for Timers
- Added: Support backend channel numbers

### CODE REVIEW

**Comment by phunkyfish** [comment]
@ksooo Now that Leia has branched once this is merged do I create another PR for master? Jsut not sure of the process. Or should I have PR'd to master first?

**Comment by ksooo** [comment]
pvr addon 'Leia' PR -> Kodi Leia
pvr addon 'Matrix' PR -> Kodi master (this is because of binary addon repo)
pvr addon 'master' PR == unused

As we're currently switching branches, please do not merge any Matrix PRs yet. Thanks.

**Comment by phunkyfish** [comment]
Sure, no problem. And when creating a PR for Matrix is it the same as for Leia or does it require the major version number to be increased?

Or just the same as Leia for the moment?

**Comment by ksooo** [comment]
I will create the first Matrix PR soon, which will bump the major number.

After that, it's the same as always. minor bump for functional improvements, micro bump if fixes only.

**Comment by ksooo** [comment]
Now that Matrix is default branch, please PR for Matrix first and submit a separate PR for backport. Leia backport will be merged only after corresponding Matrix PR.

File: src/enigma2/Epg.cpp
Comment (ksooo) [inline src/enigma2/Epg.cpp]: EPG_TAG broadcast = {0};

**Comment by phunkyfish** [comment]
So prob best to wait on Matrix PR is approved before providing the backport?

**Comment by ksooo** [comment]
> So prob best to wait on Matrix PR is approved before providing the backport?

 yep, but no need to close this one.

**Review by ksooo — APPROVED** [review]
LGTM, but please do not merge until corresponding Matrix PR got merged.

File: src/enigma2/Epg.cpp
Comment (ksooo) [reply src/enigma2/Epg.cpp]: maybe you should change that...

Comment (phunkyfish) [reply src/enigma2/Epg.cpp]: Done

**Comment by phunkyfish** [comment]
@ksooo I added one more commit, had it on a branch and it never made it into a PR. Same has been added to Matrix branch. All ready for when you are.

**Review by ksooo — APPROVED** [review]
Go ahead. :-)

**Pull request merged**
